tests/interfaces/test_jacobian_products.py

```diff
@@ -62,6 +70,70 @@ def test_transform_jacobian_product_basics(self):
         )
         assert repr(jpc) == expected_repr
 
+    def test_device_jacobians_initialization_new_dev(self):
+        """Tests the private attributes are set during initialization of a DeviceJacobians class."""
+
+        device = qml.device("default.qubit")
+        config = qml.devices.ExecutionConfig(gradient_method="adjoint")
+
+        jpc = DeviceJacobians(device, {}, config)
+
+        assert jpc._device is device
+        assert jpc._execution_config is config
+        assert jpc._gradient_kwargs == {}
+        assert jpc._is_new_device
```

Comment:
iirc there is a recommendation in pylint to make these tests explicitly check identity with `True` and `False` :thinking: 
```suggestion
        assert jpc._is_new_device is True
```